nipype/algorithms/confounds.py

```diff
@@ -249,9 +252,11 @@ class FramewiseDisplacement(BaseInterface):
     }]
 
     def _run_interface(self, runtime):
-        mpars = np.loadtxt(self.inputs.in_plots)  # mpars is N_t x 6
+        mpars = np.loadtxt(self.inputs.in_file)  # mpars is N_t x 6
         diff = mpars[:-1, :] - mpars[1:, :]
         diff[:, :3] *= self.inputs.radius
+        if self.inputs.parameter_source == "AFNI":
+            diff[:, :3] *= (np.pi / 180)
         fd_res = np.abs(diff).sum(axis=1)
```

Comment:
UGH. Do not use the above, which re-sorts the parameters. This can be done more simply by taking the last 6 columns.

```Python
mpars = np.loadtxt(self.inputs.in_file)[:, -6:]  # mpars is N_t x 6
if self.inputs.parameter_source in ('AFNI', 'FSFAST'):
    mpars[:, :3] *= np.pi / 180
diff = np.diff(mpars, axis=0)
diff[:, :3] *= self.inputs.radius
fd_res = np.abs(diff).sum(axis=1)
```

And this doesn't handle SPM, contra my previous comment, because SPM puts its rotation parameters first. (Though my confused comments may support the idea of solving the "load normalized motion params" issue once and reusing.)